graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,317 @@
+            "E-step".
+
+        optimal_transport_eps : float (default=0.1), optional
+            Tolerance for the each iteration of the iterative optimal transport
+            problem. I.e. tolerance of the whole "EM" algorithm.
+
+        iterative_num_reps : int (default=100), optional
+            Number of reps in each iteration of the iterative optimal transport
+            problem. I.e. maxumum number of total iterations the whole "EM"
+            algorithm.
+
+        init : string, {"2d" (default), "sign_flips", "custom"}, optional
+
+            - "2d"
+                uses 2^d different initiazlizations, where d is the dimension.
+                specifically, uses all possible matrices with all entries real
+                and diagonal entries having magnitude 1 and 0s everywehre else.
+                for example, for d=2, tries [[1, 0], [0, 1]], [[1, 0], [0,
+                -1]], [[-1, 0], [0, 1]], and [[-1, 0], [0, -1]]. picks the best
+                one based on the value of the objective function.
+            - 'sign_flips'
+                for the first dimension, if two datasets have medians with
+                varying signs, flips all signs along this dimension for the
+                first dataset. then initializes to an identity.
+            - "custom"
+                expects either an initial matrix Q or initial matrix P during
+                the use of fit or fit_transform. uses initial Q provided,
+                unless it is not provided. if not provided - uses initial P. if
+                neither is given initializes to Q = I.
+
+        initial_Q : np.ndarray, shape (d, d) or None, optional (default=None)
+            An initial guess for the alignment matrix, if such exists. Ignored
+            if init alignment is set to anything other than 'custom'.
+            If None - initializes using an initial guess for P.
+            If None, and P is also None - initializes Q to identity matrix.
+
+        initial_P : np.ndarray, shape (n, m) or None, optional (default=None)
+            Initial guess for the initial transport matrix.
+            Only matters if Q=None.
+
+    Attributes
+    ----------
+        Q_ : array, size (d, d)
+              final orthogonal matrix, used to modify X.
+
+        P_ : array, size (n, m) where n and md are the sizes of two datasets
+             final matrix of optimal transports
+
+    References
+    ----------
+    .. [1] Agterberg, J.
+    """
+
+    def __init__(
+        self,
+        optimal_transport_lambda=0.1,
+        optimal_transport_eps=0.01,
+        optimal_transport_num_reps=1000,
+        iterative_eps=0.01,
+        iterative_num_reps=100,
+        init="2d",
+        initial_Q=None,
+        initial_P=None,
+    ):
+        # check optimal_transport_lambda argument
+        if type(optimal_transport_lambda) is not float:
+            msg = "optimal_transport_lambda must be a float, not {}".format(
```

Comment:
Perhaps a tedious task, but change all strings to f strings instead of using `.format`? Might be worth while to encourage f strings in any future additions.

Reply:
wat

Reply:
Do `"optimal_transport_lambda must be a float, not {type(optimal_transport_lambda)}"`. You did this in L185 and L197.

Reply:
with due respect, Jaewon, i do not want to. this is extremely tedious, since there isn't even a proper way to search and replace them, whether within a file or through repo, and we use .format literally in almost every module, and this makes no practical difference anyway.

Reply:
Thats fine. FYI, the reason we did `.format` is because we allowed 3.5 at the start, which didnt support f strings.

Reply:
i did figure that out; and i'd be happy to only use fstrings from now on / from the start of this PR, have i known. but my point is: i don't want to be spending time tracking these now. if i messed up and followed the wrong convention - i can see how that would be my fault, so i'd do that, but i actually just used what seemed to be most wide-spread convention in graspy, which, to the best of my knowledge isn't planned to be depreciated by Python for next few years at least.

and btw, we're not being consistent on sooo many other things that are user-facing (e.g. take a closer look at how docs in different modules specify which value does the argument have by default) that tracking down and fixing something like this, which can't even be seen outside of source, seems like an inefficient way to spend time rn. imho.

Reply:
Im not entirely sure what triggered this tirade, but my original comment was a suggestion that could potentially improve readability of source code (i personally like f strings). The **only** reason it even crossed my mind and suggested it is because you did both styles in this. And at no point did I mean to say `.format` was the wrong convention and was your fault for using it.

Again, its fine if you dont want to change them, and we can leave it at that.